forgerock-auth/src/androidTest/java/org/forgerock/android/auth/callback/DeviceBindingCallbackTest.java

```diff
@@ -274,6 +222,58 @@ public void onCallbackReceived(Node node)
         Assert.assertNotNull(FRSession.getCurrentSession());
         Assert.assertNotNull(FRSession.getCurrentSession().getSessionToken());
     }
+
+    @Test
+    public void testDeviceBindingApplicationPin() throws ExecutionException, InterruptedException {
+        final int[] bindSuccess = {0};
+        CallbackFactory.getInstance().register(CustomApplicationPinDeviceBindingCallback.class);
+
+        ActivityScenario scenario = ActivityScenario.launch(DummyActivity.class);
+        scenario.onActivity(InitProvider::setCurrentActivity);
+
+        NodeListenerFuture<FRSession> nodeListenerFuture = new DeviceBindingNodeListener(context, "pin")
+        {
+            NodeListener<FRSession> nodeListener = this;
+            @Override
+            public void onCallbackReceived(Node node)
+            {
+                if (node.getCallback(CustomApplicationPinDeviceBindingCallback.class) != null) {
+                    CustomApplicationPinDeviceBindingCallback callback = node.getCallback(CustomApplicationPinDeviceBindingCallback.class);
+                    callback.getDeviceAuthenticator().pin = "1234";
+
+                    callback.bind(context, new FRListener<Void>() {
+                        @Override
+                        public void onSuccess(Void result) {
+                            bindSuccess[0]++;
+                            try {
+                                // Get the kid
+                                KID = JWTParser.parse((String) callback.getInputValue(0)).getHeader().toJSONObject().get("kid").toString();
```

Comment:
I don't think you want to set the global KID here, this may affect other test cases.

Reply:
Oh, definitely! moreover, this test doesn't need to validate the kid at all...